src/enzo/DetermineSEDParameters.C

```diff
@@ -0,0 +1,290 @@
+    smdot=SS->AccretionRate[SS->TimeIndex]*MassUnits/TimeUnits;
+    dt_enzo=Time-ptime;
+
+  
+
+    if(ptime>0.0){
+      FORTRAN_NAME(stellar)(&smdot, &dt_enzo, &slum, &rad);
+      SS->LuminosityPerSolarMass = slum/ParticleMass;
+    }
+    else{
+      SS->LuminosityPerSolarMass = 1e10;
+    }
+  }
+  ptime = Time;
+  return SUCCESS;
+#endif
+  /* 
+   * The PopIII values are taken from Schaerer et al. 2002 Table 4.
+   * Luminosity is NOT in ergs/s but in photons/s 
+   */
+  if(SS->ParticleClass == POPIII) {
+    SS->RadiationLifetime = SmartStarSMSLifetime*3.154e7/TimeUnits; //Code Time
+    SS->LuminosityPerSolarMass = 6.696798e49/40.0; //In physical units
+    /* Ideally we call SLUG here. Hardcoded for now */
+    for(int bin = 0; bin < NUMRADIATIONBINS; bin++) {
+      SS->RadiationEnergyBins[bin] = PopIIIEnergyBins[bin];
+      SS->RadiationSED[bin] = PopIIISEDFracBins[bin];
+    }
+  }
+  /*
+   * For the SMS spectrum I assume a blackbody with an effective 
+   * temperature of 6000K. 
+   * Luminosity is NOT in ergs/s but in photons/s 
+   */
+  else if (SS->ParticleClass == SMS) {
+    SS->RadiationLifetime = SmartStarSMSLifetime*3.154e7/TimeUnits; //Code Times
+    SS->LuminosityPerSolarMass = 1.4e51/500.0;
+    /* Ideally we call SLUG here. Hardcoded for now */
+    for(int bin = 0; bin < NUMRADIATIONBINS; bin++) {
+      SS->RadiationEnergyBins[bin] = SMSEnergyBins[bin];
+      SS->RadiationSED[bin] = SMSSEDFracBins[bin];
+    }
+  }
+  
+  /*
+   * The BH luminosity depends on the Black Hole mass and the accretion rate
+   * The values are calculated assuming a multi-colour disc for the 
+   * for accretion disk and a corona fit by a power law.
+   * The values are hardcoded into the table above and valid for black
+   * hole masses from 1e0 to 1e9 and accretion rates from 1e-7 to 1e2
+   */
+  else if(SS->ParticleClass == BH && SmartStarBHRadiativeFeedback == TRUE) {
+    SS->RadiationLifetime = 1e14*3.154e7/TimeUnits; //code time
+    double accrate = (SS->AccretionRate[SS->TimeIndex]*(MassUnits/SolarMass)/TimeUnits)*3.154e7; //Msolar/yr
+    double BHMass = ParticleMass;
+    float epsilon = SS->eta_disk;
+    double eddrate = 4*M_PI*GravConst*BHMass*SolarMass*mh/(epsilon*clight*sigma_thompson); // g/s
+    eddrate = eddrate*3.154e7/SolarMass; //in Msolar/yr
+    accrate = max(accrate, 1e-6);
+    BHMass = min(BHMass, 1.0);
+    int arrayindex = CalculateArrayIndex(BHMass, accrate);
+    SS->LuminosityPerSolarMass = BHArray[arrayindex][0]/BHMass; //erg/s/msun
+    
+    if(SmartStarSuperEddingtonAdjustment == TRUE) {
+      if(accrate > eddrate) {
+	//printf("%s: Super Eddington Accretion rate detected (%f Medd) - need to modify feedback!!!!!!!!!\n", __FUNCTION__, accrate/eddrate);
+	float mue = 1.22, a = 0.5;
+	double Ledd = 4*M_PI*GravConst*BHMass*SolarMass*mh*mue*clight/sigma_thompson; //cgs
+	double medddot = 16.0*Ledd/(clight*clight); //cgs
+	double accrate_cgs = accrate*SolarMass/3.154e7; 
```

Comment:
Replace the hard-coded year in seconds with "yr_s" from phys_const.def (also appears throughout this file).

Reply:
Thanks - good catch! Done